src/components/ComparisonPointFieldArray/AgreementField.js

```diff
@@ -159,20 +159,26 @@ class AgreementField extends React.Component {
     </div>
   )
 
-  renderError = () => (
-    <Layout className={`textCentered ${css.error}`}>
-      <strong>
-        {this.props.meta.error}
-      </strong>
-    </Layout>
-  )
+  renderError = () => {
+    const { input: { name }, meta: { error } } = this.props;
+    return (
+      <Layout className={`textCentered ${css.error}`}>
+        <strong>
+          {error?.[name]}
+        </strong>
+      </Layout>
+    );
+  }
 
   render() {
     const {
       id,
+      input: { name },
       meta: { error, touched }
     } = this.props;
 
+    const relevantError = error?.[name];
```

Comment:
Can we have a different name for this error instead of `relevantError`?

And any reason you use the `[name]` here?

Reply:
I'm happy to rename it, what do you propose? I was using `[name]` because instead of being passed a boolean like I expected, I was passed an object `{onDate: {...}}` when the (completely separate) onDate field had an error, so I was trying to filter to only the relevantError, hence the name. I just put in a fake validator though, to verify this and realised that when the package itself has an error, and the date field has an error, we instead get `{onDate: {...}, comparisonPointValidationError}`, so I'll change this code to just strip out the onDate?

Reply:
I can also move this out to utilities, since it's shared code between the two fields, package/agreement, if you think that's worthwhile

Reply:
As I suggested below, if you use form validations for this you dont need to filter out the errors....```name={`${name}[${index}].onDate`}``` which you already have in place should automatically populate the error for you and block the form from submission

Reply:
Yep, turns out the way I was dealing with fieldNames was flawed. This has been fixed and is much cleaner now.